lib/model/model.go

```diff
@@ -2572,6 +2636,22 @@ func (m *model) checkFolderRunningLocked(folder string) error {
 	return errFolderNotRunning
 }
 
+// lockBoth locks both fmut and pmut, thus ensuring that always happens in the
+// same order to prevent deadlocks. It doesn't matter in which order the locks
+// are released.
+func (m *model) lockBoth(fmutRead, pmutRead bool) {
```

Comment:
Not a fan. The multiple-boolean-parameters thing makes it hard to see at the call site what it means (no clue unless reading the signature) and I'm not sure the order thing has been a source of issues for a long time... If you think it really is then I would prefer `lockFandRLockP()` or whatever as more explicit. (Or consolidate both to a single lock for less trickiness to begin with.)